File: sort/shaker_sort/go/shaker_sort_test.go
package shakersort

import (
	"sort"
	"testing"
)

func TestSort(t *testing.T) {
	dataset := []float32{2, 7, 3, 9, 2, 5, 4, 1, 92, 12, 45, 22, 56, 24, 6.2}
	sorted := ShakerSort(dataset)
	sortedAscending := func(i, j int) bool {
		return sorted[i] < sorted[j]
	}
	isCorrectlySorted := sort.SliceIsSorted(sorted, sortedAscending)
	if isCorrectlySorted == false {
		t.Errorf("Failed to sort in ascending order using shaker sort algorithm.")
	}
}

File: sort/shaker_sort/php/shaker_sort_test.php
<?php

include "shaker_sort.php";

$array = [42, 5, 89, 2, 44, 0, 23, -2, 109, 76, 19, -6, 99, 98];
$sorted = shaker_sort($array);

$expected = [-6, -2, 0, 2, 5, 19, 23, 42, 44, 76, 89, 98, 99, 109];
echo ($expected == $sorted) ? 'Array is sorted !!' : 'Array is not sorted as expected';
var_dump('array', $array, 'sorted set', $sorted);